src/app/components/authorization/authorization.tsx

```diff
@@ -0,0 +1,153 @@
+
+  const registerHendlerRef = useRef<any>(null);
+  const loginHendlerRef = useRef<any>(null);
+
+  const registerEmailRef = useRef<any>(null);
+  const registerPasswordRef = useRef<any>(null);
+  const loginEmailRef = useRef<any>(null);
+  const loginPasswordRef = useRef<any>(null);
+
+  const navigate = useNavigate();
+  const register = async () => {
+    try {
+      const user = await createUserWithEmailAndPassword(
+        auth,
+        registrationEmail,
+        registrationPassword
+      );
+      const currentUser:any = auth.currentUser;
+      sendEmailVerification(currentUser);
+      auth.signOut();
+      alert("Email Send");
+      navigate("/");
+    } catch (error: any) {
+      // console.error(error.message);
+      if (error.message) {
+        setAuthorizationError("Registration Error");
+      }      
+    }
+  }
+
+  const login = async () => {   
+    try {
+      const UserCredential = await signInWithEmailAndPassword(
+        auth,
+        loginEmail,
+        loginPassword
+      );
+      navigate("/");
+    } catch (error: any) {
+      // console.error(error.message);
+      if (error.message) {
+        setAuthorizationError("Incorrect username or password");
+      }
+    }
+  }
+
+  return (
+    <Box className="moviebase-authorization-block">
+      <img src={bigLogo} alt="big-logo" className='big-logo'/>
+      <Box className='moviebase-authorization-card'>
+        <Box className="card-header-block">
+          <div className='card-header-register moviebase-authorization-card-active' 
+          onClick={() => {            
+            setAuthorizationHenadler("register");
+            setAuthorizationError("");
+            setRegistrationEmail("");
+            setRegistrationPassword("");
+            registerHendlerRef.current.classList.add("moviebase-authorization-card-active");
+            loginHendlerRef.current.classList.remove("moviebase-authorization-card-active");          
+          }} 
+          ref={registerHendlerRef}>Registration</div>
+          <div className='card-header-login' 
+          onClick={() => { 
+            setAuthorizationHenadler("login");
+            setAuthorizationError("");
+            setLoginEmail("");
+            setLoginPassword("");
+            loginHendlerRef.current.classList.add("moviebase-authorization-card-active");
+            registerHendlerRef.current.classList.remove("moviebase-authorization-card-active");
+          }}
+          ref={loginHendlerRef}
+          >Login</div>
+        </Box>
+        {authorizationHendler === "register" ? 
+        <Box className="card-registration-block">
+          <TextField
+            label="Email"
+            variant="standard"
+            id="card-registration-email"
+            autoComplete="off"
+            value={registrationEmail}
+            ref={registerEmailRef}
+            onChange={e => setRegistrationEmail(e.target.value)}
+          />
+    
+          <TextField
```

Comment:
For your experience, better to use some form library like this one - https://formik.org/